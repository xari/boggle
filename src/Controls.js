import React from "react";
import { Switch } from "@headlessui/react";
import { classNames } from "./utils";
import "./Controls.css";

function Range({ value, onChange }) {
  return (
    <div className="my-2">
      <span className="flex-grow flex flex-col">
        <label className="text-md font-medium text-gray-900">Board size</label>
        <span className="text-sm text-gray-500">
          Slide the bar to resize the board.
        </span>
      </span>
      <input
        type="range"
        min="4"
        max="6"
        step="1"
        value={value}
        onChange={onChange}
      />
    </div>
  );
}

function BoardSwitch({ enableRandom, setEnableRandom }) {
  return (
    <Switch.Group as="div">
      <span className="flex-grow flex flex-col">
        <Switch.Label
          as="span"
          className="text-md font-medium text-gray-900"
          passive
        >
          Use auto-generated board
        </Switch.Label>
        <Switch.Description as="span" className="text-sm text-gray-500">
          Toggle betwenn an auto-generated board and a customizable one.
        </Switch.Description>
      </span>
      <Switch
        checked={enableRandom}
        onChange={setEnableRandom}
        className={classNames(
          enableRandom ? "bg-indigo-600" : "bg-gray-200",
          "my-2 relative inline-flex flex-shrink-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
        )}
      >
        <span
          aria-hidden="true"
          className={classNames(
            enableRandom ? "translate-x-5" : "translate-x-0",
            "pointer-events-none inline-block h-5 w-5 rounded-full bg-white shadow transform ring-0 transition ease-in-out duration-200"
          )}
        />
      </Switch>
    </Switch.Group>
  );
}

export default function Controls({
  dimensions,
  setDimensions,
  enableRandom,
  setEnableRandom,
}) {
  return (
    <>
      <Range value={dimensions} onChange={setDimensions} />
      <BoardSwitch
        enableRandom={enableRandom}
        setEnableRandom={setEnableRandom}
      />
    </>
  );
}
