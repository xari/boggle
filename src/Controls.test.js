import React from "react";
import renderer from "react-test-renderer";
import Controls from "./Controls";

describe("the Controls", () => {
  it("render without any unexpected changes", () => {
    const tree = renderer
      .create(<Controls dimensions={4} enableRandom={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
