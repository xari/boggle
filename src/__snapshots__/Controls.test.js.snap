// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Controls render without any unexpected changes 1`] = `
Array [
  <div
    className="my-2"
  >
    <span
      className="flex-grow flex flex-col"
    >
      <label
        className="text-md font-medium text-gray-900"
      >
        Board size
      </label>
      <span
        className="text-sm text-gray-500"
      >
        Slide the bar to resize the board.
      </span>
    </span>
    <input
      max="6"
      min="4"
      step="1"
      type="range"
      value={4}
    />
  </div>,
  <div>
    <span
      className="flex-grow flex flex-col"
    >
      <span
        className="text-md font-medium text-gray-900"
        id="headlessui-label-1"
      >
        Solve an auto-generated board
      </span>
      <span
        className="text-sm text-gray-500"
        id="headlessui-description-2"
      >
        Toggle betwenn an auto-generated board and a customizable one.
      </span>
    </span>
    <button
      aria-checked={true}
      aria-describedby="headlessui-description-2"
      aria-labelledby="headlessui-label-1"
      className="bg-indigo-600 my-2 relative inline-flex flex-shrink-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
      id="headlessui-switch-3"
      onClick={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      role="switch"
      tabIndex={0}
      type="button"
    >
      <span
        aria-hidden="true"
        className="translate-x-5 pointer-events-none inline-block h-5 w-5 rounded-full bg-white shadow transform ring-0 transition ease-in-out duration-200"
      />
    </button>
  </div>,
]
`;
